blueflood-core/src/main/java/com/rackspacecloud/blueflood/rollup/Granularity.java

```diff
@@ -254,23 +280,25 @@ private static Granularity granularityFromPointsLinear(double requestedDuration,
      *
      * @param requestedDuration (milliseconds)
      */
-    private static Granularity granularityFromPointsGeometric(String tenantid, long from, long to, double requestedDuration, int requestedPoints) {
+    private static Granularity granularityFromPointsGeometric(String tenantid, long from, long to, double requestedDuration, int requestedPoints, long assumedIntervalMillis, boolean checkTtl) {
         double minimumPositivePointRatio = Double.MAX_VALUE;
         Granularity gran = null;
-        if (SAFETY_TTL_PROVIDER == null) {
+        if (checkTtl && SAFETY_TTL_PROVIDER == null) {
             SAFETY_TTL_PROVIDER = SafetyTtlProvider.getInstance();
         }
 
```

Comment:
I am leaning towards agreeing with @ChandraAddala . I think all 3 of us agree that the existing implementation will need to be changed slightly to accommodate testing. The challenge is how do we change the existing implementation so that we leave the behavior/logic of the class as in-tact as possible.

Using the ```Clock``` seems reasonable. @ChandraAddala had refactored ```ScheduledContext``` this way. He added a constructor that takes an instance of Clock:
```
public ScheduleContext(long currentTimeMillis, Collection<Integer> managedShards)
public ScheduleContext(long currentTimeMillis, Collection<Integer> managedShards, Clock clock)
```
Our normal services code will use the first constructor, which will use a default ```Clock``` implementation. Our test code can mock ```Clock``` and can use the second constructor. This leaves the remaining logic of the code in-tact.

Adding ```checkTtl``` parameter to ```granularityFromPointsInInterval()``` changes the logic of the method. It opens up a different behavior of the method, which is to skip checking TTL. I don't think that's good.